libs/sdk-ui-dashboard/src/presentation/saveAs/DefaultSaveAsDialog/index.tsx

```diff
@@ -1,8 +1,48 @@
-// (C) 2019-2020 GoodData Corporation
-import React from "react";
+// (C) 2019-2022 GoodData Corporation
+import React, { useCallback } from "react";
 import { ISaveAsDialogProps } from "../types";
 import { useSaveAs } from "./useSaveAs";
 import { SaveAsDialogRenderer } from "./SaveAsDialogRenderer";
+import { useToastMessage } from "@gooddata/sdk-ui-kit";
+import {
+    selectIsSaveAsDialogOpen,
+    uiActions,
+    useDashboardDispatch,
+    useDashboardSelector,
+} from "../../../model";
+
+/**
+ * @internal
+ */
+export function useSaveAsDialogProps(): ISaveAsDialogProps {
+    const { addSuccess, addError } = useToastMessage();
+
+    const dispatch = useDashboardDispatch();
+    const closeSaveAsDialog = () => dispatch(uiActions.closeSaveAsDialog());
```

Comment:
This is created on every invocation but in the onSaveAsError only uses the first version. Should this also be a useCallback?